src/mock/MockGraph.ts

```diff
@@ -44,21 +67,42 @@ export class MockGraph extends BaseGraph {
     // The MockGraph isn't making real Graph requests, so we can simply no-op and return the same instance.
     return this;
   }
+}
 
+/**
+ * Implements Middleware for the Mock Client to escape
+ * the graph url from the request
+ *
+ * @class MockMiddleware
+ * @implements {Middleware}
+ */
+// tslint:disable-next-line: max-classes-per-file
+class MockMiddleware implements Middleware {
   /**
-   * get events for Calendar
-   *
-   * @param {Date} startDateTime
-   * @param {Date} endDateTime
-   * @returns {Promise<MicrosoftGraph.Event[]>}
-   * @memberof MockGraph
+   * @private
+   * A member to hold next middleware in the middleware chain
    */
-  public async getEvents(startDateTime: Date, endDateTime: Date): Promise<MicrosoftGraph.Event[]> {
-    const sdt = `startdatetime=${startDateTime.toISOString()}`;
-    const edt = `enddatetime=${endDateTime.toISOString()}`;
-    const uri = `/me/calendarview?${sdt}&${edt}`;
+  private _nextMiddleware: Middleware;
+
+  // tslint:disable-next-line: completed-docs
+  public async execute(context: Context): Promise<void> {
+    try {
+      const url = context.request as string;
+      const baseLength = BASE_URL.length;
+      context.request = url.substring(0, baseLength) + escape(url.substring(baseLength));
 
-    const calendarView = await this.client.api(escape(uri)).get();
-    return calendarView ? calendarView.value : null;
+      return await this._nextMiddleware.execute(context);
+    } catch (error) {
+      throw error;
```

Comment:
Is this doing anything? Seems odd to catch an error just to rethrow it.

Reply:
lol, good catch, this is a copy and paste error. Fixed it in both places